### PR TITLE
1487600: Cockpit - Save configuration from register dialog

### PR DESCRIPTION
**Pull request by jirihnidek**
* Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=1487600
* Try to save all settings from register dialog
* Do not try to auto-attach, when registration failed
* TODO: auto-attach can fail, when proxy server is used
* Automaticaly request update of page, when dialog is closed

### CODE REVIEW

File: cockpit/src/subscriptions-client.js [hunk 1]
```diff
@@ -248,63 +266,72 @@ client.registerSystem = subscriptionDetails => {
...
+                                    configService.Set('server.' + key, connection_options[dict[key]])
+                                        .catch(error => {
+                                            console.error('unable to save server.' + key, error);
+                                        });
+                                }
+                            } else {
+                                // When config option was not specified in dialog, then it is
+                                // necessary to reset it in rhsm.conf (proxy options are optional)
+                                console.debug('resetting: server.' + key);
+                                    configService.Set('server.' + key, {'t': 's', 'v': ''})
```
Comment (kahowell) [inline cockpit/src/subscriptions-client.js]: Looks like if you don't specify port or prefix, then they will get reset by this logic. I think the logic should reset to default... you could either pull this values for now from the defaults in `subscriptions-client.js`, or, ideally, we could pull them from the service which would require some enhancement to the d-bus service... I am fine with just pulling from hardcoded values for now with a TODO.

Comment (jirihnidek) [reply cockpit/src/subscriptions-client.js]: @kahowell Thanks for review. Good point. I found another case during testing :-). Please create another Trello card for desired D-Bus changes.

**Comment by kahowell** [comment]
Card for adding defaults to the d-bus service: https://trello.com/c/riWH2CS1

**Comment by jirihnidek** [comment]
@kahowell Thanks for hint. I fixed this and one more warning of jshint.

**Pull request merged**